components/board/genericlinux/data.go

```diff
@@ -129,76 +128,74 @@ func readIntFile(filePath string) (int, error) {
 	return int(resultInt64), err
 }
 
-func getGpioChipDefs(pinDefs []PinDefinition) (map[string]gpioChipData, error) {
-	gpioChipsInfo := map[string]gpioChipData{}
-	sysfsPrefixes := []string{"/sys/devices/", "/sys/devices/platform/", "/sys/devices/platform/bus@100000/"}
+func getGpioChipDefs(pinDefs []PinDefinition) (map[int]gpioChipData, error) {
+	gpioChipsInfo := map[int]gpioChipData{}
 
-	// Get a set of all the chip names with duplicates removed. Go doesn't have native set objects,
-	// so we use a map whose values are ignored.
-	gpioChipNames := make(map[string]struct{}, len(pinDefs))
-	for _, pinDef := range pinDefs {
-		if pinDef.GPIOChipSysFSDir == "" {
-			continue
-		}
-		gpioChipNames[pinDef.GPIOChipSysFSDir] = struct{}{}
-	}
+	allDevices := gpio.ChipDevices()
+	gpioChipNgpios := make(map[int]string, len(allDevices)) // maps chipNgpio -> string gpiochip#
+	for _, dev := range allDevices {
+		chip, _ := gpio.OpenChip(dev)
```

Comment:
You're ignoring an error here, which could lead to trouble. We should check it instead.
```go
chip, err := gpio.OpenChip(dev)
if err != nil {
    return nil, err
}
chipInfo, err := chip.Info()
...
```